src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -112,8 +112,9 @@ internal unsafe MsQuicStream(MsQuicConnection.State connectionState, SafeMsQuicS
 
             // Inbound streams are already started
             _state.StartCompletionSource.SetResult();
-
             _state.Handle = streamHandle;
+            _streamId = GetStreamId(); // needs _state.Handle to be set
```

Comment:
We could make it static and pass the handle to GetStreamId(). But this is fine for now.